src/components/PeopleTable.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { useParams } from 'react-router-dom';
+import cn from 'classnames';
+
+interface Props {
+  people: Person[];
+}
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+
+  const getPersonByName = (name: string | null) =>
+    people.find(person => person.name === name);
+
+  const getParentTableData = (person: Person, forMother: boolean) => {
+    if (forMother) {
```

Comment:
You are doing the same in both cases, just use different keys to get data from object

as a  second argument, pass this key:
`const getParentTableData = (person: Person, parentKey: 'motherName' | 'fatherName')`

and use it to get data from object